include/ignition/gui/qml/PlottingInterface.qml

```diff
@@ -0,0 +1,546 @@
+              main.window.show();
+          }
+      }
+  }
+  Component {
+      id : exportWindow
+      ApplicationWindow {
+          id: exportApp
+          width: 1000; height: 600
+          title: "Plotting Export"
+
+          property var currentChart: null
+          property bool isSmallChart: false
+          property bool isCopyFinished: false
+
+          property var charts: []
+          property int index: 0
+
+          property var chartImages: []
+
+          function addChartCopy(result)
+          {
+              var imageObject = chartCopy.createObject(grid);
+              imageObject.setSource(result.url);
+
+              imageObject.setChartName("chart" + charts[index].chartID.toString());
+              imageObject.chartIndex = index;
+
+              chartImages.push(imageObject);
+
+              if (isSmallChart)
+              {
+                charts[index].width = 200;
+                charts[index].height = 135;
+                charts[index].multiChartsMode = true;
+                charts[index].fillPlotInOrOut();
+                isSmallChart = false;
+              }
+
+              index++;
+              continueCopy();
+          }
+
+          function setCharts(_charts)
+          {
+              // clear
+              charts = []
+              chartImages = []
+
+              Object.keys(_charts).forEach(function(key) {
+                  charts.push(_charts[key]);
+              });
+
+              continueCopy();
+          }
+
+          function continueCopy()
+          {
+
+              if (charts[index].multiChartsMode)
+              {
+                charts[index].getChart().animationOptions = ChartView.NoAnimation
+                charts[index].width  = 1270
+                charts[index].height = 720
+                charts[index].multiChartsMode = false;
+                charts[index].fillPlotInOrOut();
+                isSmallChart = true;
+              }
+
+              if (index < charts.length)
+                charts[index].getChart().grabToImage(addChartCopy);
+          }
+
+
+          Text {
+              id : ray2;
+              anchors.centerIn: parent
+              text: "RRRR"
+          }
+
+          function exportCSV(path)
+          {
+            for (var i = 0; i < chartImages.length; i++)
+            {
+                if (!chartImages[i].isSelected())
+                    continue;
+
+                var chart = charts[chartImages[i].chartIndex];
+                var serieses = chart.getChart().getAllSerieses();
+                var chart_id = chart.chartID;
+
+                if (Object.keys(serieses).length === 0)
+                    continue;
+
+
+                // convert Serieses to Map of {series_name : points list}
+                // slots in cpp accepts QMap<QString, QVariant>
+                // QVariant could be a list instead of series object
+
+                var chartSerieses = {};
+                let seriesArray = [];
```

Comment:
what's `let`, typo?

Reply:
what is "typo" ? :D 
in this code i put the series's points into a list "seriesArray", as slots in cpp will not accept the series object itself

Reply:
I had to change the `let` to `var` for the plugin to load. It looks like Qt only suppors ES6 (which introduces `let`) from version 5.12, but on Ubuntu Bionic we're on Qt 5.9.5.